zerver/tests/test_middleware.py

```diff
@@ -32,32 +29,19 @@ def test_is_slow_query(self) -> None:
         self.assertFalse(is_slow_query(9, '/accounts/webathena_kerberos_login/'))
         self.assertTrue(is_slow_query(11, '/accounts/webathena_kerberos_login/'))
 
-    @override_settings(SLOW_QUERY_LOGS_STREAM="logs")
-    @patch('logging.info')
-    def test_slow_query_log(self, mock_logging_info: Mock) -> None:
-        error_bot = get_system_bot(settings.ERROR_BOT)
-        create_stream_if_needed(error_bot.realm, settings.SLOW_QUERY_LOGS_STREAM)
-
-        self.log_data['time_started'] = time.time() - self.SLOW_QUERY_TIME
-        write_log_line(self.log_data, path='/socket/open', method='SOCKET',
-                       remote_ip='123.456.789.012', requestor_for_logs='unknown', client_name='?')
-        last_message = self.get_last_message()
-        self.assertEqual(last_message.sender.email, "error-bot@zulip.com")
-        self.assertIn("logs", str(last_message.recipient))
-        self.assertEqual(last_message.topic_name(), "testserver: slow queries")
-        self.assertRegexpMatches(last_message.content,
-                                 r"123\.456\.789\.012 SOCKET  200 10\.\ds .*")
-
-    @override_settings(ERROR_BOT=None)
-    @patch('logging.info')
-    @patch('zerver.lib.actions.internal_send_stream_message')
-    def test_slow_query_log_without_error_bot(self,
-                                              mock_internal_send_stream_message: Mock,
-                                              mock_logging_info: Mock) -> None:
+    def test_slow_query_log(self) -> None:
         self.log_data['time_started'] = time.time() - self.SLOW_QUERY_TIME
-        write_log_line(self.log_data, path='/socket/open', method='SOCKET',
-                       remote_ip='123.456.789.012', requestor_for_logs='unknown', client_name='?')
-        mock_internal_send_stream_message.assert_not_called()
+        with patch("zerver.middleware.slow_query_logger") as mock_slow_query_logger, \
+                patch("zerver.middleware.logger") as mock_normal_logger:
+
+            write_log_line(self.log_data, path='/socket/open', method='SOCKET',
```

Comment:
SOCKET isn't a method we have anymore; let's change it to the normal parameters for sending a message.

Reply:
Changed